### ISSUE TITLE
can't use square brackets in docs (conflict with link syntax)

### ISSUE THREAD

**Issue by atacratic** [unisonweb/unison#2306]
Both these docs fail to parse:

```
{{ hello [remark] }}

{{ a type like `[i] -> [o]` }}
```
with the following error pointing to the character after the closing `]`
```
  unexpected space
  expecting '('
```
I guess it's a conflict with link syntax, as described here: https://www.unisonweb.org/docs/documentation#link-syntax-cheat-sheet

It would be good if square brackets could be used.  

Sorry if this is a dup, I do half-remember seeing it mentioned somewhere.

**Comment by rlmark** [comment]
@atacratic Yeah the first case should probably be supported. If you just want monospace formatting for the latter case I've found that the following will work: 

```
myDoc = {{
    a type like ''[i] -> [o]'' 
}}
```

(The way to monospace in this version of the docs is two successive single quotes.)

**Comment by atacratic** [comment]
I think the second case should be supported as well - needed when you want to talk about code that doesn't necessarily parse or work, or code in language other than unison.  
```
{{ this is similar to the haskell `foo :: [i] -> [o]` }}
```

**Comment by pchiusano** [comment]
I think if we can't support single backtick as the syntax, we gotta produce a nice warning / error when users write docs with a single backtick and tell them what to do instead to produce monospaced text.

For the first case, where you want to write something that conflicts with existing syntax, you can do:

```
doc = {{ This is a {{ docWord "[remark]" }} }}
```

This works for any escaping you'd like to do.

For the second case, if it's inline monospaced text, surround in double single quotes as @rlmark said, ex: `''[Int] -> [Int]''` and whatever is inside won't be parsed as a doc.

There's also block monospaced - 

````
``` haskell
-- front end may attempt syntax highlighting
foo :: Int -> Int
```
OR
'''
some monospace text, not syntax highlighted
'''
````
